### PR TITLE
Change in TestInbound() test, where TestUntil is replaced with Access…

### PR DESCRIPTION
**Pull request by mihaj**
…Safely

Linked to #48

### CODE REVIEW

File: src/Vlingo.Wire.Tests/Fdx/Inbound/MockInboundStreamInterest.cs [hunk 1]
```diff
@@ -30,15 +30,14 @@ public void HandleInboundStreamMessage(AddressType addressType, RawMessage messa
             var textMessage = message.AsTextMessage();
             TestResult.Messages.Add(textMessage);
             TestResult.MessageCount.IncrementAndGet();
-            Debug.WriteLine($"INTEREST: {textMessage} list-size: {TestResult.Messages.Count} count: {TestResult.MessageCount.Get()} count-down: {TestResult.UntilStops.Remaining}");
-            TestResult.UntilStops.Happened();
```
Comment (tjaskula) [inline src/Vlingo.Wire.Tests/Fdx/Inbound/MockInboundStreamInterest.cs]: You cannot just do that. You have to signal it has happened with something like `TestResult.UntilStops.WriteUsing(<name>, 1);` Look on how this is implemented in `MockChannelReaderConsumer`

File: src/Vlingo.Wire.Tests/Fdx/Inbound/MockInboundStreamInterest.cs [hunk 2]
```diff
@@ -30,15 +30,14 @@ public void HandleInboundStreamMessage(AddressType addressType, RawMessage messa
             var textMessage = message.AsTextMessage();
             TestResult.Messages.Add(textMessage);
             TestResult.MessageCount.IncrementAndGet();
-            Debug.WriteLine($"INTEREST: {textMessage} list-size: {TestResult.Messages.Count} count: {TestResult.MessageCount.Get()} count-down: {TestResult.UntilStops.Remaining}");
```
Comment (tjaskula) [inline src/Vlingo.Wire.Tests/Fdx/Inbound/MockInboundStreamInterest.cs]: You don't need to get rid of that one. You can pass into the constructor the expected number and calculate it substracting `TestResult.UntilStops.TotalWrites`

File: src/Vlingo.Wire.Tests/Fdx/Inbound/InboundStreamTest.cs [hunk 3]
```diff
@@ -26,7 +26,7 @@ public class InboundStreamTest: IDisposable
         [Fact]
         public void TestInbound()
         {
-            _interest.TestResult.UntilStops = TestUntil.Happenings(1);
+            _interest.TestResult.UntilStops = AccessSafely.AfterCompleting(1);
```
Comment (tjaskula) [inline src/Vlingo.Wire.Tests/Fdx/Inbound/InboundStreamTest.cs]: you need to configure something like 

```
var count= 0;
            var accessSafely = AccessSafely.AfterCompleting(4)
                .WritingWith<int>("count", (value) => count += value)
                .ReadingWith("count", () => count);
 _interest.TestResult.UntilStops = accessSafely ;
```

File: src/Vlingo.Wire.Tests/Fdx/Inbound/InboundStreamTest.cs [hunk 4]
```diff
@@ -38,9 +38,7 @@ public void TestInbound()
                 ++count;
                 Assert.Equal(MockChannelReader.MessagePrefix + count, message);
             }
-   
-            _interest.TestResult.UntilStops.Completes();
```
Comment (tjaskula) [inline src/Vlingo.Wire.Tests/Fdx/Inbound/InboundStreamTest.cs]: `_interest.TestResult.UntilStops.ReadFromExpecting("count", 1);`

File: src/Vlingo.Wire.Tests/Fdx/Inbound/InboundStreamTest.cs [hunk 5]
```diff
@@ -16,31 +17,35 @@
...
             int count = 0;
-            foreach (var message in _interest.TestResult.Messages)
-            {
-                ++count;
-                Assert.Equal(MockChannelReader.MessagePrefix + count, message);
-            }
-   
-            _interest.TestResult.UntilStops.Completes();
-    
+            count += (from item in Enumerable.Range(1, _interest.TestResult.Messages.Count)
```
Comment (tjaskula) [inline src/Vlingo.Wire.Tests/Fdx/Inbound/InboundStreamTest.cs]: We're not using anywhere in the code base this format of linq queries. I would advice either leaving the loop which is more readable in that particular case or use the fluent format `_interest.TestResult.Messages.Where(...prefix).Count()`

I think also that you can join the declaration and assignment like `var count = <query>`

Comment (mihaj) [reply src/Vlingo.Wire.Tests/Fdx/Inbound/InboundStreamTest.cs]: Yeah, I will covert it to the loop, query LINQ syntax does not fit in.

Comment (mihaj) [reply src/Vlingo.Wire.Tests/Fdx/Inbound/InboundStreamTest.cs]: @tjaskula I've checked the tests and this was the only one using `TestUntil`. A quick search showed that TestUntil is set in the `TestRequestChannelConsumerProvider` class, but it's not referenced anywhere. Should I remove it?

Comment (tjaskula) [reply src/Vlingo.Wire.Tests/Fdx/Inbound/InboundStreamTest.cs]: > @tjaskula I've checked the tests and this was the only one using `TestUntil`. A quick search showed that TestUntil is set in the `TestRequestChannelConsumerProvider` class, but it's not referenced anywhere. Should I remove it?

Thanks @mihaj. Could you just get rid of the `TestUntil` property in `TestRequestConsumerProvider` ?

**Pull request merged**